blogposts/inside-rust/01-announcement.md

```diff
@@ -53,8 +53,8 @@ no plans to change this.
 Platform ABIs can define a special kind of unwinding called "forced unwinding."
 This type of unwinding operation cannot be stopped prematurely using language
 mechanisms such as Rust's `catch_unwind` or C++'s `catch`. (These mechanisms
-can intercept forced unwinds, but the runtime will resume unwinding regardless
-of the content of the intercepting code, unless the program is aborted).
+can sometimes intercept forced unwinds, but they must resume the unwind
+ otherwise the runtime will abort the program).
```

Comment:
Looking at the spec, it seems there is no suggestion of automatically aborting:

> A catch-all block may be executed during forced unwinding. For instance, a `longjmp may execute code in a catch(...) during stack unwinding. However, if this happens, unwinding will proceed at the end of the catch-all block, whether or not there is an explicit rethrow. 

....but I suppose a compiler could automatically insert some logic to abort rather than letting the unwind be resumed.

I'm trying to determine whether GCC (and by extension, maybe C++ in general?) have the behavior you're describing, where a non-rethrowing `catch` block would `abort` instead of automatically resuming the unwind. But on my Mac, it seems like I don't see destructors being run whether or not I rethrow the `pthread_exit` exception. In fact, I don't even see a print statement in the `catch` block itself. http://coliru.stacked-crooked.com/a/ee07fca96fe69c47

Reply:
Basically only glibc on Linux uses forced unwinding for pthread_exit. Even musl on Linux just exits the thread directly without unwinding.

The abort happens because glibc defines the destructor of the exception object passed to _Unwind_ForcedUnwind to abort if the exception object is ever destroyed (which happens if you catch it without rethrowing it).

Reply:
Ah, that makes a lot of sense. It seems like it's probably worth noting that the statement about `pthread_exit` only applies to glibc on Linux, though.